pkg/apiserver/handlers.go

```diff
@@ -72,13 +72,35 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+type LongRunningRequestCheck func(r *http.Request) bool
+
+// BasicLongRunningRequestCheck pathRegex operates against the url path, the queryParams match is case insensitive.
+// Any one match flags the request
+func BasicLongRunningRequestCheck(pathRegex *regexp.Regexp, queryParams map[string]string) LongRunningRequestCheck {
+	return func(r *http.Request) bool {
```

Comment:
Originally we discussed implementing this inside the restful stack and
having a catchall for everything outside, so the stack would have access to
APIRequestInfo and potentially the request.Scope.  It's probably too much
work to do now, but we should queue it up for 1.3 (in fact, I bet there is
an issue I opened over a year ago on this).

On Thu, Feb 18, 2016 at 7:48 PM, David Eads <notifications@github.com>
wrote:

> In pkg/apiserver/handlers.go
> <https://github.com/kubernetes/kubernetes/pull/21497#discussion_r53408004>
> :
>
> > @@ -72,13 +72,35 @@ func ReadOnly(handler http.Handler) http.Handler {
> >  	})
> >  }
> >
> > +type LongRunningRequestCheck func(r *http.Request) bool
> > +
> > +// BasicLongRunningRequestCheck pathRegex operates against the url path, the queryParams match is case insensitive.
> > +// Any one match flags the request
> > +func BasicLongRunningRequestCheck(pathRegex *regexp.Regexp, queryParams map[string]string) LongRunningRequestCheck {
> > +	return func(r *http.Request) bool {
>
> What prevents me from adding ?watch=true to all of my requests to avoid
> the rate limiter?
>
> Nothing. How much logic do we want to put in here to inspect the request?
> It's possible to run a RequestInfo parse and make a more definitive
> determination.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/21497/files#r53408004>.
>